src/mon/MonClient.cc

```diff
@@ -1136,11 +1136,13 @@ void MonClient::handle_get_version_reply(MMonGetVersionReply* m)
 
 AuthAuthorizer* MonClient::build_authorizer(int service_id) const {
   Mutex::Locker l(monc_lock);
-  assert(auth || active_con->get_auth());
-  if (auth)
+  if (auth) {
     return auth->build_authorizer(service_id);
-  else
-    return active_con->get_auth()->build_authorizer(service_id);
```

Comment:
You don't explain in the commit comment why you no longer try active_con->get_auth().  It sounds like you were just replacing the assert with a message.